File: 2. Sorting/SelectionSort.py
def SelectionSort(lst, n):

    for i in range(0, n-1):
        
        min_flag=i

        for j in range(i+1, n):

            if(lst[j]<lst[min_flag]):
                min_flag=j;

        if(min_flag!=i):
            lst[min_flag], lst[i]= lst[i], lst[min_flag]

arr = [64, 34, 25, 12, 22, 11, 90]

n = len(arr)

SelectionSort(arr, n)

print(arr)

File: 2. Sorting/SelectionSort.cpp
#include <iostream>
#include <stdio.h>
#include <conio.h>

void swap(int &a, int &b)
{
    int c = a;
    a = b;
    b = c;
}

int SelectionSort(int arr[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {

        int min_flag=i;

        for (int j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[min_flag])
            {
                min_flag=j;
            }
        }

        if(min_flag!=i)
        {
            swap(arr[min_flag], arr[i]);
        }
    }
}

int main()
{
    system("cls");

    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);

    SelectionSort(arr, n);

    for (int i = 0; i <= n-1; i++)
    {
        std::cout << arr[i] << " ";
    }

    return 0;
}